pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -450,3 +451,40 @@ func (a Adapter) Delete(labels map[string]string) error {
 
 	return a.Client.DeleteDeployment(labels)
 }
+
+func (a Adapter) Log(follow, debug bool) (io.ReadCloser, error) {
+
+	if !utils.ComponentExists(a.Client, a.ComponentName) {
+		return nil, errors.Errorf("the component %s doesn't exist on the cluster", a.ComponentName)
+	}
+
+	var command versionsCommon.DevfileCommand
+	var err error
+
+	if debug {
+		command, err = common.GetDebugCommand(a.Devfile.Data, "")
+		if err != nil {
+			return nil, err
+		}
+		if reflect.DeepEqual(versionsCommon.DevfileCommand{}, command) {
+			return nil, errors.Errorf("no debug command found in devfile, please run \"odo logs\" for run command logs")
+		}
+
+	} else {
+		command, err = common.GetRunCommand(a.Devfile.Data, "")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	containerName := command.Exec.Component
+
+	// Wait for Pod to be in running state otherwise we can't sync data or exec commands to it.
+	pod, err := a.waitAndGetComponentPod(true)
```

Comment:
We can use https://github.com/openshift/odo/blob/e367c334c2e6a15f584821c34af4659bb2658a30/pkg/kclient/pods.go#L153
Waiting might take a long time. I think it's better to error out by checking the status of the returned pod and error out saying `component is not running on the cluster`

Reply:
updated